docs/tutorials/developer-journey/level-3/3.2-https-outcalls.md

```diff
@@ -0,0 +1,813 @@
+    public type TransformRawResponseFunction = {
+        function : shared query TransformArgs -> async HttpResponsePayload;
+        context : Blob;
+    };
+
+    // This Type defines the arguments the transform function needs.
+    public type TransformArgs = {
+        response : HttpResponsePayload;
+        context : Blob;
+    };
+
+    public type CanisterHttpResponsePayload = {
+        status : Nat;
+        headers : [HttpHeader];
+        body : [Nat8];
+    };
+
+    public type TransformContext = {
+        function : shared query TransformArgs -> async HttpResponsePayload;
+        context : Blob;
+    };
+
+
+    // Lastly, declare the IC management canister which we use to make the HTTPS outcall.
+    public type IC = actor {
+        http_request : HttpRequestArgs -> async HttpResponsePayload;
+    };
+
+}
+```
+
+The code above is annotated with detailed notes and explanations. Take a moment to read through the code's content and annotations to fully understand the code's functionality. 
+
+Now, we can deploy the canister locally with the commands:
+
+```
+dfx start --background --clean
+dfx deploy
+```
+
+:::caution
+Recall that if you'd like to deploy this canister on the mainnet, you can use the `--network ic` flag. 
+:::
+
+These commands should return the canister's URL. Open the `http_get_backend` canister's URL to open the Candid UI:
+
+```
+  Backend canister via Candid interface:
+    http_get_backend: http://127.0.0.1:4943/?canisterId=a3shf-5eaaa-aaaaa-qaafa-cai&id=avqkn-guaaa-aaaaa-qaaea-cai
+```
+
+### Calling the HTTP `GET` request
+
+Once you open the Candid UI, you can use the 'Call' button to call the `get_icp_usd_exchange()` method, which sends an HTTP `GET` request. The result will be returned in the UI, such as:
+
+![HTTP GET](../_attachments/http_get_candid.png)
+
+You can play with a version of this canister running on-chain here: https://a4gq6-oaaaa-aaaab-qaa4q-cai.raw.ic0.app/?id=fm664-jyaaa-aaaap-qbomq-cai
+
+## HTTPS `POST`
+
+To demonstrate how to use the HTTPS `POST` outcall, we'll create a simple canister that has one public method named `send_http_post_request()`. This method will trigger an HTTP `POST` request to a public API service where the HTTPS request can be inspected and we can inspect the request's headers and body to validate that it reflects what the canister sent. 
+
+The following are some important notes on `POST` requests:
+
+- Since HTTPS outcalls go through consensus, it is expected that any HTTPS `POST` request may be sent several times to it's destination. This is because it is common for clients to retry requests for a variety of reasons.
+
+- It is recommended that HTTP `POST` requests add the idempotency keys in the header of the request so that the destination knows which `POST` requests from the client are identical. 
+
+- Developers should be cautious using idempotency keys since the destination server may not know how to understand and use them. 
+
+### Prerequisites
```

Comment:
it looks like everything below this line is duplicated from up above

Reply:
The Prerequisites and Creating a New Project  portions were duplicated, but the code is different. You make a good point that its redundant to duplicate these, so I've revised the page to eliminate the duplicated steps and dive right into the `POST` code.